### ISSUE TITLE
pip-sync does not use --extra-index-url

### ISSUE THREAD

**Issue by stefanfoulis** [jazzband/pip-tools#192]
`pip-sync` does not discover package versions from `--extra-index-url`.

`pip-sync requirements.txt` with the following `requirements.txt`:

```
--extra-index-url https://example.com/django-backports/+simple/
Django==1.6.11.post1
```

causes:

```
Collecting django==1.6.11.post1
  Could not find a version that satisfies the requirement django==1.6.11.post1 (from versions: 1.1.3, 1.1.4, 1.2, 1.2.1, 1.2.2, 1.2.3, 1.2.4, 1.2.5, 1.2.6, 1.2.7, 1.3, 1.3.1, 1.3.2, 1.3.3, 1.3.4, 1.3.5, 1.3.6, 1.3.7, 1.4, 1.4.1, 1.4.2, 1.4.3, 1.4.4, 1.4.5, 1.4.6, 1.4.7, 1.4.8, 1.4.9, 1.4.10, 1.4.11, 1.4.12, 1.4.13, 1.4.14, 1.4.15, 1.4.16, 1.4.17, 1.4.18, 1.4.19, 1.4.20, 1.4.21, 1.5, 1.5.1, 1.5.2, 1.5.3, 1.5.4, 1.5.5, 1.5.6, 1.5.7, 1.5.8, 1.5.9, 1.5.10, 1.5.11, 1.5.12, 1.6, 1.6.1, 1.6.2, 1.6.3, 1.6.4, 1.6.5, 1.6.6, 1.6.7, 1.6.8, 1.6.9, 1.6.10, 1.6.11, 1.7, 1.7.1, 1.7.2, 1.7.3, 1.7.4, 1.7.5, 1.7.6, 1.7.7, 1.7.8, 1.7.9, 1.8a1, 1.8b1, 1.8b2, 1.8rc1, 1.8, 1.8.1, 1.8.2, 1.8.3)
  Some externally hosted files were ignored as access to them may be unreliable (use --allow-external django to allow).
No matching distribution found for django==1.6.11.post1
```

`pip install -r requirements.txt` works fine.

**Comment by nvie** [comment]
Correct, this is a missing feature. `pip-sync` should support it. Any volunteers for implementing this?

**Comment by chrisbarr** [comment]
+1 - this would be really useful for us :)

**Comment by davidovich** [comment]
Fixed in #283
Closing.
